apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/parser/graphql/GraphQLDocumentParser.kt

```diff
@@ -531,7 +531,8 @@ class GraphQLDocumentParser(
             fragmentRefs = fragmentRefs.union(mergeInlineFragmentRefs).toList(),
             inlineFragments = inlineFragments.result.filter { it.typeCondition != typeCondition },
             filePath = graphQLFilePath,
-            sourceLocation = SourceLocation(start)
+            sourceLocation = SourceLocation(start),
+            packageName = "${packageNameProvider.operationPackageName(graphQLFilePath)}.fragment".removePrefix(".")
```

Comment:
A few comments:

* What about dropping the ".fragment" here? As long as all fragments ended up in the same package but if we start using the file package name instead, it might be confusing to append ".fragment
* Since fragments are not operations, we might want to rename `operationPackageName` to `filePackageName`
* It's ok to change this behaviour for experimental multi-module things but not for the stable non-multi-module users so this will have to be configurable somewhere.

Reply:
Thanks for the feedback:
- I'm for dropping the `fragment` package suffix. 
- Makes sense, however, if we want to make this behavior configurable, maybe having a separate `fragmentPackageName()` function that encapsulates this logical fork?
- Do you have a suggestion for a flag name?

Reply:
>I'm for dropping the fragment package suffix.

👍 

>Do you have a suggestion for a flag name?

Maybe something like: 

```
apollo {
  // Use the path to the file that defined the fragment as package name
  // i.e. com/example/Fragment.graphql will end up in `com.example` package instead of a global `*.fragment` package
  useFilePackageNameForFragments.set(true)
}
```

> maybe having a separate fragmentPackageName() function that encapsulates this logical fork?

Yup, that works too